### ISSUE TITLE
Trying to decode JSON for raw_file

### ISSUE THREAD

**Issue by agimenez** [bluefeet/GitLab-API-v4#21]
When using the raw_file method, GitLab returns the raw contents, but the REST client will try to decode JSON from it anyway.

I have found line https://github.com/bluefeet/GitLab-API-v4/blob/9cbb00821eb413fdc231a52b069f9b753325c8b3/lib/GitLab/API/v4/RESTClient.pm#L179 to be what generates the error.

In the same `if` block, I can see that there is a `decode` option that can't be specified from the client, but that option is only set internally from the v4.pm module (https://github.com/bluefeet/GitLab-API-v4/blob/9cbb00821eb413fdc231a52b069f9b753325c8b3/lib/GitLab/API/v4.pm#L6734).

I think that for `raw_file`, we should not try to decode the response (as we are just requesting raw contents, and can be anything other than JSON), so I guess we should unset the decode option, similar to the following?:

```perl
sub raw_file {
    my $self = shift;
    croak 'raw_file must be called with 2 to 3 arguments' if @_ < 2 or @_ > 3;
    croak 'The #1 argument ($project_id) to raw_file must be a scalar' if ref($_[0]) or (!defined $_[0]);
    croak 'The #2 argument ($file_path) to raw_file must be a scalar' if ref($_[1]) or (!defined $_[1]);
    croak 'The last argument (\%params) to raw_file must be a hash ref' if defined($_[2]) and ref($_[2]) ne 'HASH';
    my $params = (@_ == 3) ? pop() : undef;
    my $options = {};
    $options->{query} = $params if defined $params;
    $options->{decode} = 0; # <--------- THIS
    return $self->_call_rest_client( 'GET', 'projects/:project_id/repository/files/:file_path/raw', [@_], $options )
}
```

Does that make sense?

**Comment by agimenez** [comment]
Hi @bluefeet did you have a chance to take a look at this?

Thank you very much!

**Comment by bluefeet** [comment]
Ya, taking a look now.  The old {decode} flag was from a previously botched change that actually didn't need to disable decoding.  But I left the innner-workings in in case there would be a need, and, low-and-behold, this looks like it'll be the case.

I need to make sure the whole set of repository file methods work, especially create file and update file so I can write a full end-to-end test against a live server.  Otherwise I won't have a reproducible way to test this and I often find some other broken bits getting the tests working, so raises the test coverage.

Also will need to adjust the code generator to accept a NODECODE option for particular API endpoints.

So may not be getting this fixed today, but I'm on it.

**Comment by agimenez** [comment]
Yeah, that's what I thought, it couldn't be that easy, that's why I didn't do a PR. 

Just tell me if you need some help with this.

Thanks!

**Comment by bluefeet** [comment]
Aight, this seems to do it well: 1b42725403e01cbcb17b37928bfaff7daf8ff23a
If you could, double check the diff on `lib/GitLab/API/v4.pm` - its pretty simple in the end.

**Comment by bluefeet** [comment]
And I changed my mind again.  :)

See 7832025604f5682004bc155b306fe109ca046827.

**Comment by bluefeet** [comment]
Fixed in 0.11.
